structured.js

```diff
@@ -505,12 +506,53 @@
             }
             // Recursively check for matches
             if ((_.isArray(currTree[key]) &&
-                    checkNodeArray(currTree[key], toFind, peersToFind, wVars, matchResults, options)) ||
+                    checkNodeArray(currTree[key], toFind, peersToFind, wVars, matchResults, options, false)) ||
                 (!_.isArray(currTree[key]) &&
-                    checkMatchTree(currTree[key], toFind, peersToFind, wVars, matchResults, options))) {
+                    checkMatchTree(currTree[key], toFind, peersToFind, wVars, matchResults, options, false))) {
                 return matchResults;
             }
         }
+        if (!recursing) {
```

Comment:
Hm. It seems strange to me that this function seems to act recursively when recursing is false. Can you add comments that further explain how this works?